src/main/java/com/linecorp/armeria/server/docs/ThriftXmlDocString.java

```diff
@@ -0,0 +1,217 @@
+        StringBuilder sb = new StringBuilder();
+        if (packageName != null) {
+            sb.append(packageName).append('.'); // '.' is FQCN's delimiter.
+        }
+        int count = aggregatedNames.size();
+        int i = 1;
+        for (String name : aggregatedNames) {
+            sb.append(name);
+            if (i < count) {
+                sb.append(DELIM);
+            }
+            i++;
+        }
+        return sb.toString();
+    }
+
+    static Map<String, String> getDocStringsFromXmlResource(ClassLoader classLoader, String xmlResourcePath) {
+        final InputStream in = classLoader.getResourceAsStream(
+                ObjectUtil.checkNotNull(xmlResourcePath, "xmlResourcePath"));
+        if (in == null) {
+            return null;
+        }
+
+        final Deque<String> fullName = new LinkedList<>();
+        final Map<String, String> docStrings = new HashMap<>();
+        String packageName = null;
+        XMLStreamReader reader = null;
+
+        try {
+            reader = XML_INPUT_FACTORY.createXMLStreamReader(in);
+            while (reader.hasNext()) {
+                int readerStatus = reader.next();
+                if (readerStatus == XMLEvent.START_ELEMENT) {
+                    final Map<String, String> attrs = attrsToMap(reader);
+                    final String tagName = reader.getLocalName();
+                    if ("namespace".equals(tagName)) {
+                        if (packageName == null && "java".equals(attrs.get("name"))) {
+                            packageName = attrs.get("value");
+                        }
+                    }
+                    if (DOC_CONTAINABLE_XML_ELEMENTS.contains(tagName)) {
+                        final String key = tagName.equals("throws") ? "type-id" : "name";
+                        final String name = attrs.get(key);
+                        final String doc = attrs.get("doc");
+                        if (name == null) {
+                            throw new IllegalArgumentException("not found '" + key + "' of " + tagName);
+                        }
+                        fullName.add(name);
+                        if (doc != null) {
+                            docStrings.put(join(packageName, fullName), doc);
+                        }
+                    }
+                } else if (readerStatus == XMLEvent.END_ELEMENT) {
+                    if (DOC_CONTAINABLE_XML_ELEMENTS.contains(reader.getLocalName())) {
+                        fullName.removeLast();
+                    }
+                }
+            }
+            return docStrings;
+        } catch (XMLStreamException e) {
+            throw new IllegalStateException(e);
+        } finally {
+            if (reader != null) {
+                try {
+                    reader.close();
+                } catch (XMLStreamException ignored) {}
+            }
+            if (in != null) {
```

Comment:
`close()`'s signature is `public void close() throws IOException`. If we use try-with-resources here, try-with-resources' `IOException` is not covered.